src/test/java/com/google/api/codegen/testdata/library_gapic.yaml

```diff
@@ -49,6 +49,12 @@ interfaces:
     init_fields:
       - name%shelf="testShelf"
       - name%book="testBook"
+  iam_resources:
+  - type: google.example.library.v1.Book
+    field: name
+  iam_replace_methods:
+    - AddTag
+    - AddLabel
```

Comment:
This is not the right level of abstraction for GAPIC config; our config should not be aware of domain-specific concerns (of which IAM is an example). A more appropriate level of abstraction is mix-ins. For example, I added the `reroute_to_grpc_interface` to handle the pubsub mix-in hack. It looks like for Go, you want to ignore that rerouting and use a different mechanism. We should be careful about "settings clusters" where some settings are used by some languages but a mutually-exclusive set of settings are used by others. We have this for flattening vs required fields; in that case, the connection is well-enough understood, and they sit side-by-side. In your case, there is no apparent connection. 

If we are supporting a domain-modeled approach here, we should define the settings in terms of that concept. Potentially we should be explicit in config that this only applies to Go. 

* `iam_replace_methods`: potentially merge with `reroute_to_grpc_interface` to create a new abstraction. Example:

```
  - name: AddLabel
    ...
    surface_treatments:
    - exclude_languages: go
      reroute_to_grpc_interface: google.tagger.v1.Labeler
    - include_languages: go
      visibility: disabled
```

* Potentially bootstrap config related to generating accessors to handles like you need:
```
interfaces:
- name: google.example.library.v1.LibraryService
  ...
  handles:
  - include_languages: go
    input:
    - name: book
      type: google.example.library.v1.Book
    passthrough:
    // This should be understood in a language-independent way; "self" refers to the api wrapper object.
    - name: self.connection
    // dotted accessor path
    - name: book.name
    output:
    - type: google.cloud.iam.Handle
```